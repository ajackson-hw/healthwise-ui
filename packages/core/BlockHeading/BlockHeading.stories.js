import React from 'react'
import { storiesOf } from '@storybook/react'
import BlockHeading from './index'

storiesOf('core|Components/Block Heading', module).add(
  'with text content',
  () => <BlockHeading>Sample Content</BlockHeading>,
  {
    info: 'Demonstates basic usage with text content'
  }
)
