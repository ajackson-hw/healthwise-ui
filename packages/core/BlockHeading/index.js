import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './BlockHeading.css'

class BlockHeading extends Component {
  render() {
    const headingClasses = classNames({
      'hw-block-heading': true,
      [styles.block_heading]: true,
    })
    const textClasses = classNames({
      'hw-block-heading-text': true,
      [styles.block_heading_text]: true,
    })

    return (
      <h3 className={headingClasses}>
        <span className={textClasses}>{this.props.children}</span>
      </h3>
    )
  }
}

BlockHeading.propTypes = {
  children: PropTypes.node.isRequired,
}

export default BlockHeading
