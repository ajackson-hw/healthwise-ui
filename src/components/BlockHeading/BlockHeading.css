.block_heading {
  box-sizing: border-box;
  width: 100%;
  background-color: var(--color-primary);
  text-align: left;
  padding: 0.5em 24px;
  margin: 1em 0 0.5em;
  font-weight: normal;
  font-size: 1.5em;
  line-height: 1.55em;
}

.block_heading_text {
  box-sizing: border-box;
  color: var(--color-text-on-primary);
  width: 100%;
}

@media print {
  .block_heading {
    background: #f5f5f5;
    margin: 4px 0 12px;
    border: 1px solid #ccc;
    padding: 0;
  }

  .block_heading_text {
    padding: 5px 16px;
    margin: 0;
    font-size: 14px;
    color: #000;
  }
}
