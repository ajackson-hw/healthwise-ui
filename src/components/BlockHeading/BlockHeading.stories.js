import React from 'react'
import { storiesOf } from '@storybook/react'
import BlockHeading from './index'

storiesOf('Components/Block Heading', module).addWithInfo(
  'with text content',
  `
      Demonstates basic usage with text content
    `,
  () => <BlockHeading>Sample Content</BlockHeading>
)
