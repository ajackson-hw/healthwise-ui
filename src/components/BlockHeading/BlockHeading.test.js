/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import BlockHeading from './index'

describe('<BlockHeading />', () => {
  it('renders correctly with text node child', () => {
    const component = renderer.create(<BlockHeading>Block Heading</BlockHeading>).toJSON()

    expect(component).toMatchSnapshot()
  })
})
