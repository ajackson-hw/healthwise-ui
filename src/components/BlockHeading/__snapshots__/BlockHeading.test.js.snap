// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockHeading /> renders correctly with text node child 1`] = `
<h3
  className="hw-block-heading block_heading"
>
  <span
    className="hw-block-heading-text block_heading_text"
  >
    Block Heading
  </span>
</h3>
`;
